aws_lambda_powertools/shared/user_agent.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import os
+
+from importlib_metadata import version
```

Comment:
Hey Roger, I'll check this PR tomorrow.

Just a quick note: please add this package as a main dependency on Poetry. We need to move it from the dev dependency, as the Lambda Runtime doesn't come with this package by default.

```sh
poetry remove importlib-metadata --dev
poetry add  importlib-metadata

```
![image](https://github.com/awslabs/aws-lambda-powertools-python/assets/4295173/d2081063-ad6a-409a-b12e-41b9368b0946)